modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpTaskState.java

```diff
@@ -206,12 +206,27 @@ public static Metadata fromXContent(XContentParser parser) {
         }
 
         public boolean isCloseToExpiration() {
-            return Instant.ofEpochMilli(lastCheck).isBefore(Instant.now().minus(25, ChronoUnit.DAYS));
+            final Instant now = Instant.ofEpochMilli(System.currentTimeMillis()); // millisecond precision is sufficient (and faster)
```

Comment:
I think we could pass in a ThreadPool, and use ThreadPool::absoluteTimeInMillis, which would be significantly faster since it's cached. Same goes in the other place below.

Reply:
That makes sense. I'll see if I can make that happen.

Reply:
I can't make it pass tests! I'm going to open a new ticket to track it.